tests/rules/assertions/checkTagNames.js

```diff
@@ -1,5 +1,19 @@
 /* eslint-disable no-restricted-syntax */
 
+const JSDOC_3_TAGS = [
```

Comment:
Please list them one value per line.

Since this would make the file very long, it might make sense to list these values in a JSON file and just import the JSON file.